dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -51,6 +51,178 @@ public static void prepare() throws Exception {
         rulesAPI   = APILocator.getRulesAPI();
     }
 
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveConditionGroup(ConditionGroup, User, boolean)}
+     *                  and {@link RulesAPIImpl#saveCondition(Condition, User, boolean)}
+     * When: User with permission try to create a Group Condition and a Condition
+     * Should: Delete it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveCondition() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule = new RuleDataGen().host(host).nextPersisted();
+
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
+        this.addPermission(role, host, true);
+
+        //Saving and testing GroupCondition
+        rulesAPI.saveConditionGroup(conditionGroup, user, false);
+
+        List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        assertEquals(1, allRules.size());
+
+        final Rule ruleFromDataBase = allRules.get(0);
+
+        final List<ConditionGroup> groups = ruleFromDataBase.getGroups();
+        assertEquals(1, groups.size());
+
+        assertEquals(groups.get(0).getOperator(), conditionGroup.getOperator());
+
+        //Saving and testing Condition
+        final Condition condition = new ConditionDataGen().group(groups.get(0)).next();
+
+        rulesAPI.saveCondition(condition, user, false);
+
+        allRules = rulesAPI.getAllRules(user, false);
+        assertEquals(1, allRules.size());
```

Comment:
this assert doesn't look like it's needed

Reply:
done https://github.com/dotCMS/core/pull/18127/commits/d8a32ec5a775f434c7ee34caeaa3b0e19018ab5f#diff-9baded4fb0fb63e2648d9c9c4380ec1aL93